test/pkg.jl

```diff
@@ -15,6 +15,20 @@ include("utils.jl")
 
 const TEST_PKG = (name = "Example", uuid = UUID("7876af07-990d-54b4-ab0e-23690620f79a"))
 
+import Pkg3.Types.semver_spec
+@testset "semver notation" begin
+    @test semver_spec("^1.2.3") == Pkg3.Types.VersionSpec("1.2.3-2")
+    @test semver_spec("^1.2")   == Pkg3.Types.VersionSpec("1.2.0-2")
+    @test semver_spec("1")      == Pkg3.Types.VersionSpec("1.0.0-2")
```

Comment:
Is this implementing the Cargo-style semver_spec where "no prefix" == "^ prefix"? If so, it might be better to test it here with the caret for clarity and then separately test that the no prefix version is equal to the `^` version.

Reply:
:+1:, done